lib/backend/src/Obelisk/Backend.hs

```diff
@@ -263,29 +312,111 @@ renderGhcjsFrontend urlEnc ghcjsWidgets route configs f = do
 
 -- | Preload all.js in a link tag.
 -- This is the default preload method.
-preloadGhcjs :: Text -> FrontendWidgetT r ()
-preloadGhcjs allJsUrl = elAttr "link" ("rel" =: "preload" <> "as" =: "script" <> "href" =: allJsUrl) blank
+preloadGhcjs :: GhcjsWasmAssets -> FrontendWidgetT r ()
+preloadGhcjs (GhcjsWasmAssets allJsUrl mWasm) = case mWasm of
+  Nothing -> elAttr "link" ("rel" =: "preload" <> "as" =: "script" <> "href" =: allJsUrl) blank
+  Just wasmAssets -> scriptTag $ view _1 (wasmScripts allJsUrl wasmAssets)
 
 -- | Load the script from the given URL in a deferred script tag.
 -- This is the default method.
-deferredGhcjsScript :: Text -> FrontendWidgetT r ()
-deferredGhcjsScript allJsUrl = elAttr "script" ("type" =: "text/javascript" <> "src" =: allJsUrl <> "defer" =: "defer") blank
+deferredGhcjsScript :: GhcjsWasmAssets -> FrontendWidgetT r ()
+deferredGhcjsScript (GhcjsWasmAssets allJsUrl mWasm) = case mWasm of
+  Nothing -> elAttr "script" ("type" =: "text/javascript" <> "src" =: allJsUrl <> "defer" =: "defer") blank
+  Just wasmAssets -> scriptTag $ view _2 (wasmScripts allJsUrl wasmAssets)
+
+scriptTag :: DomBuilder t m => Text -> m ()
+scriptTag t = elAttr "script" ("type" =: "text/javascript") $ text t
+
+wasmScripts
+  :: Text
+  -- ^ URL (could be relative) of "all.js"
+  -> (Text, WasmAssets)
+  -- ^ Root URL (could be relative) of wasm assets along with
+  -- the record of wasm assets.
+  -> (Text, Text, (Int -> Text))
+  -- ^ (preload script, deferred run script, delayed run script (Int input is the delay))
+wasmScripts allJsUrl' (wasmRoot, wAssets) = (preloadScript, runJsScript, delayedJsScript)
+  where
+    wrapName f = "'" <> wasmRoot <> "/" <> f wAssets <> "'"
```

Comment:
Since this stuff is user-supplied we should do at least basic sanitizing on it. Maybe replacing `'` with `\'` at least.

Reply:
Isn't there some library (jmacro?) that actually knows how to do this properly?  We should not be implementing escaping ourselves, and we certainly shouldn't be doing it inline in other code.